app/src/main/java/org/oppia/app/player/audio/AudioFragmentPresenter.kt

```diff
@@ -96,63 +72,38 @@ class AudioFragmentPresenter @Inject constructor(
       fragment.childFragmentManager.beginTransaction().remove(previousFragment).commitNow()
     }
     val dialogFragment = LanguageDialogFragment.newInstance(
-      languages as ArrayList<String>,
-      selectedLanguageCode
+      ArrayList(viewModel.languages),
+      viewModel.selectedLanguageCode
     )
     dialogFragment.showNow(fragment.childFragmentManager, TAG_LANGUAGE_DIALOG)
   }
 
-  /** Save selected language and language list to bundle */
-  fun handleSaveInstanceState(outstate: Bundle) {
-    outstate.putString(KEY_SELECTED_LANGUAGE, selectedLanguageCode)
-    outstate.putStringArrayList(KEY_LANGUAGE_LIST, ArrayList(languages))
-  }
-
   /** Pauses audio if in prepared state */
   fun handleOnStop() {
     if (!fragment.requireActivity().isChangingConfigurations && prepared) {
-      getAudioViewModel().pauseAudio()
+      viewModel.pauseAudio()
     }
   }
 
   /** Releases audio player resources */
   fun handleOnDestroy() {
     if (!fragment.requireActivity().isChangingConfigurations) {
-      getAudioViewModel().handleRelease()
+      viewModel.handleRelease()
     }
   }
 
-  /**
-   * Retrieves VoiceoverMapping from the ExplorationDataController
-   * Sets languages in Presenter and ViewModel, selected language code is defaulted to first
-   */
-  private fun retrieveVoiceoverMappings(explorationId: String, stateId: String) {
-    val explorationResultLiveData = explorationDataController.getExplorationById(explorationId)
-    processExplorationLiveData(explorationResultLiveData).observe(fragment, Observer {
-      val state = it.statesMap[stateId] ?: State.getDefaultInstance()
-      val contentId = state.content.contentId
-      val voiceoverMapping = (state.recordedVoiceoversMap[contentId] ?: VoiceoverMapping.getDefaultInstance()).voiceoverMappingMap
-      languages = voiceoverMapping.keys.toList()
-      selectedLanguageCode = languages.firstOrNull() ?: ""
-      val viewModel = getAudioViewModel()
-      viewModel.setVoiceoverMappings(voiceoverMapping)
-      viewModel.setAudioLanguageCode(selectedLanguageCode)
-    })
+  fun setVoiceoverMappings(explorationId: String, stateId: String, contentId: String? = null) = viewModel.setVoiceoverMappings(explorationId, stateId, contentId)
+  fun getCurrentPlayStatus() = viewModel.playStatusLiveData
```

Comment:
Nit here & elsewhere: please make sure there's 1 empty newline between function declarations for spacing.